src/modules/jcms_rest/src/Plugin/rest/resource/AbstractRestResourceBase.php

```diff
@@ -540,4 +539,38 @@ public function getEntityQueueItem(EntityInterface $node, FieldItemListInterface
     return $item_values;
   }
 
+  /**
+   * Returns the endpoint from the rest resource "canonical" annotation.
+   *
+   * @return string
+   * @throws \Exception
+   */
+  function getEndpoint(): string {
+    $r = new \ReflectionClass(static::class);
+    $annotation = $r->getDocComment();
+    preg_match("/\"canonical\" = \"(.+)\"/", $annotation, $endpoint);
+    if (!$endpoint) {
+      throw new \Exception('Canonical URI not found in rest resource.');
+    }
+    return $endpoint[1];
+  }
+
+  /**
+   * Gets the content type for the current rest resource.
+   * @todo Handle this in the response object optionally.
+   * @todo Handle versioning properly.
+   *
+   * @return string
+   * @throws \Exception
+   */
+  public function getContentType(): string {
+    $endpoint = $this->getEndpoint();
+    $mapper = new PathMediaTypeMapper();
+    $content_type = $mapper->getMediaTypeByPath($endpoint);
+    if (!$content_type) {
+      throw new \Exception('Content type not found for specified rest resource.');
+    }
+    return $content_type . ';version=1';
```

Comment:
We should create a separate ticket to allow better control of the version number to reduce technical debt.

Actually, it would be an improvement to allow the version number to be passed as a parameter with a default value of 1.

`function getContentType($version = 1): string {`